spec/lib/migrations/load_rate_reference_spec.rb

```diff
@@ -2,51 +2,74 @@
-      expect(subject.sheet(0).row(699)).to match_array(["02791", "Bristol", "No", "Rating Area 6"])
+    before :context do
+      invoke_task
     end
 
-    it "should have the right data" do
-      expect(subject.sheet(0).row(2).first.to_i).to eq RateReference.first.zip_code
-      expect(subject.sheet(0).row(2)[1]).to eq RateReference.first.county
-      expect(subject.sheet(0).row(2).last).to eq RateReference.first.rating_region
+    context "it creates RateReference elements correctly" do
+      subject { RateReference.first }
+      it_should_behave_like "a rate reference", { zip_code: "01001",
+                                                  county: "Hampden",
+                                                  multiple_counties: false,
+                                                  rating_region: "Rating Area 1"
+                                                }
+      # Which simply replaces all of these:
+      #
+      # it { is_expected.to have_attributes(zip_code: "01001") }
+      # it { is_expected.to have_attributes(county: "Hampden") }
+      # it { is_expected.to have_attributes(multiple_counties: false) }
+      # it { is_expected.to have_attributes(rating_region: "Rating Area 1") }
+      #
+      # Which is just equilavent to a series of these:
+      #
+      # it "assigns the correct zip code" do
+      #   expect(subject.zip_code).to eq('01001')
+      # end
     end
 
-    it "should not have duplicate data" do
-      sheet = subject.sheet(0)
-      rate_references = RateReference.where(zip_code: sheet.cell(2,1), county: sheet.cell(2,2), multiple_counties: to_boolean(sheet.cell(2,3)),
-        rating_region: sheet.cell(2,4))
-      expect(rate_references.size).to eq 1
+    context "it handles the case where multiple counties exist in the same Zip Code" do
+      let(:rate_references_by_zip) { RateReference.where(zip_code: "01002") }
+      context "first rate reference for zip code" do
+        subject { rate_references_by_zip.first }
+        it_should_behave_like "a rate reference", { zip_code: "01002",
+                                                    county: "Hampshire",
+                                                    multiple_counties: true,
+                                                    rating_region: "Rating Area 1"
+                                                  }
+      end
+
+      context "second rate reference for zip code" do
+        subject { rate_references_by_zip.second }
+        it_should_behave_like "a rate reference", { zip_code: "01002",
+                                                    county: "Franklin",
+                                                    multiple_counties: true,
+                                                    rating_region: "Rating Area 1"
+                                                  }
+      end
     end
 
     private
 
     def invoke_task
       Rake::Task["load_rate_reference:update_rating_regions"].invoke
     end
-
```

Comment:
This is now rendered unnecessary since the test does not need to know anything about that logic